Lib/crypt.py

```diff
@@ -19,18 +19,31 @@ def __repr__(self):
         return '<crypt.METHOD_{}>'.format(self.name)
 
 
-def mksalt(method=None, *, log_rounds=12):
+def mksalt(method=None, *, rounds=None):
     """Generate a salt for the specified method.
 
     If not specified, the strongest available method will be used.
 
     """
     if method is None:
         method = methods[0]
-    if not method.ident:
+    if not method.ident:  # traditional
         s = ''
-    elif method.ident[0] == '2':
+    elif method.ident[0] == '2':  # Blowfish variants
+        if rounds is None:
+            log_rounds = 12
+        else:
+            log_rounds = int.bit_length(rounds-1)
+            if rounds != 1 << log_rounds:
+                raise ValueError('rounds must be a power of 2')
+            if not 4 <= log_rounds <= 31:
+                raise ValueError('rounds out of the range 2**4 to 2**31')
         s = f'${method.ident}${log_rounds:02d}$'
+    elif method.ident in ('5', '6') and rounds is not None:  # SHA-2
+        range(rounds)  # raise a TypeError for non-integers
```

Comment:
This is a really weird way to check if rounds is an integer. Why not just testing isinstance(rounds, int) and raise a TypeError explicitly? The check should be done at the top of the function.